### PR TITLE
adding instructions for setting CORS headers when developing locally

### PR DESCRIPTION
**Pull request by macohen**
Signed-off-by: Mark Cohen <markcoh@amazon.com>

### Description
This adds instructions to add settings to opensearch.yml so developers can run opensearch locally and also develop an application on the same host. 

### Issues Resolved
None

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
I think this is more of a developer guide item that should go into https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/DEVELOPER_GUIDE.md

**Comment by macohen** [comment]
I think this is a good point and maybe it needs to also be linked to in the DEVELOPER_GUIDE.md for Dashboards. However, it is not specific to Dashboards because the CORS headers need to be configured on the OpenSearch Server to allow specific (or all) clients to connect to it. That's why I think it belongs either in the main documentation or maybe in https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md.

**Comment by dblock** [comment]
@macohen Is the audience the developer on OpenSearch core or OpenSearch Dashboards? Think of them as two separate products. This should go into the developer guide for the right audience.

**Comment by macohen** [comment]
> @macohen Is the audience the developer on OpenSearch core or OpenSearch Dashboards? Think of them as two separate products. This should go into the developer guide for the right audience.

The audience is OpenSearch core because I came across the issue when running core locally and building an independent web app not in OpenSearch Dashboards.

**Comment by macohen** [comment]
@Naarcha-AWS suggested that this might be best in https://opensearch.org/docs/latest/opensearch/configuration/ because it's not docker specific and does apply to OpenSearch Server. I'll undo the original change and apply the new one this week.

File: _opensearch/configuration.md [hunk 1]
```diff
@@ -85,3 +85,17 @@ compatibility.override_main_response_version: true
 ```
 
 The demo configuration includes a number of settings for the security plugin that you should modify before using OpenSearch for a production workload. To learn more, see [Security]({{site.url}}{{site.baseurl}}/security-plugin/).
+
+### (Optional) CORS header configuration
+If you are working on a client application running against an OpenSearch cluster on a different domain, you'll want to configure headers in opensearch.yml to allopw and developing an application locally as well, you will need to address [Cross Origin Resource Sharing](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) so your application can make calls to the OpenSearch API running locally.  You can add the following lines in your custom-opensearch.yml file (note that the "-" must be the first character in each line). You will also want to set this up if you are running OpenSearch inside Docker on the same machine as your web application.
```
Comment (Naarcha-AWS) [inline _opensearch/configuration.md:90]:
```suggestion
If you are working on a client application running against an OpenSearch cluster on a different domain, you can configure headers in `opensearch.yml` to allow for developing a local application on the same machine.  Use [Cross Origin Resource Sharing](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) so your application can make calls to the OpenSearch API running locally.  Add the following lines in your `custom-opensearch.yml` file (note that the "-" must be the first character in each line). 
```

**Review by Naarcha-AWS — CHANGES_REQUESTED** [review]
Looks good. Adjusted the language slightly to remove passive voice. Lastly, we'll need to add DCO sign-off to your last few commits. If you're looking for an easy out, open a new branch and a new PR. Remember to use `-s` to sign your commit messages.

**Comment by AMoo-Miki** [comment]
Can you please update this PR after rebasing `main` on to your dev branch. I see 48 files changed with many unrelated commits which doesn't sound right. If your branch continues to say # commits ahead and the number is more than the number of commits you are PRing, something is very wrong.

**Comment by Naarcha-AWS** [comment]
@macohen: Let's close this PR and open up a new branch, that we avoid any potential conflicts between this branch and main. Feel free to email me at naarcha@amazon.com if you need any help.

**Comment by macohen** [comment]
Closing as per @Naarcha-AWS. Will create a fresh branch, re-apply the change, and open a new PR.

**Pull request closed without merging**